src/components/form.tsx

```diff
@@ -34,7 +36,28 @@ function Form({ addPersonCard }: IFormProps) {
     register,
     clearErrors,
     formState: { errors },
+    getValues,
+    setValue,
   } = useForm<IFormValues>({ reValidateMode: 'onSubmit' });
+  const { globalState, dispatch } = useContext(GlobalStateContext);
+
+  useEffect(() => {
+    return () => {
+      if (!isFirstFilling || !isSubmitDisabled) {
```

Comment:
I highly don't recommend to relay on `useEffect` especially for business logic:
1. It is hard to read because it is too imperative
2. You should be careful with dependencies to not trigger redundant logic
Instead, you can use `onSubmit` callback on the form to run the logic